portal/dict_tools.py

```diff
@@ -30,7 +30,7 @@ def dict_match(newd, oldd, diff_stream):
 
     """
     added, removed, modified, same = dict_compare(newd, oldd)
-    if len(same) == len(newd.keys()) and len(newd.keys()) == len(oldd.keys()):
+    if len(same) == len(list(newd.keys())) and len(list(newd.keys())) == len(list(oldd.keys())):
```

Comment:
I imagine the list(*.keys()) came from http://python-future.org/compatible_idioms.html#dict-keys-values-items-as-a-list, but I don't believe that's necessary to check the `len`gth?  (That is, we don't need to force it to be of type list, just comparing the length of the two here)

Reply:
Looks like you're right. I added the list(*.keys()) with 2to3 but I see now that they're not necessary in this case.